src/backend/utils/indexer.js

```diff
@@ -7,7 +7,58 @@ const { client } = require('../lib/elastic');
 const { logger } = require('./logger');
 
 const index = 'posts';
-const type = 'post';
+
+// creates 'posts' index and add settings and mapping for autocomplete
+const setupPostsIndex = async () => {
+  try {
+    const response = await client.indices.exists({ index });
+    if (response.statusCode === 404) {
+      await client.indices.create({
+        index,
+        body: {
+          settings: {
+            analysis: {
+              analyzer: {
+                autocomplete_analyzer: {
+                  tokenizer: 'autocomplete',
+                  filter: ['lowercase', 'remove_duplicates'],
```

Comment:
If you already have an analyzer that removes duplicate, do you need a filter in `search.js`? Both should be ok.

Reply:
Its not the same thing. 
This is for removing duplicate `tokens` in the same position when indexing so it doesn't bloat up memory
https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-remove-duplicates-tokenfilter.html